realm/realm-library/src/main/java/io/realm/RealmFieldType.java

```diff
@@ -45,15 +48,29 @@
     DOUBLE(10),
     OBJECT(12),
     LIST(13),
-    LINKING_OBJECTS(14);
+    LINKING_OBJECTS(14),
+
+    @SuppressWarnings("PointlessArithmeticExpression")
+    INTEGER_LIST(LIST_OFFSET + 0),
```

Comment:
I believe `BINARY` is BLOB actually so it should probably be `BINARY_LIST`. I'm not sure but perhaps `byte` is store in `INTEGER` columns. You are right about `RealmList<byte[]>`. I'm not sure if core supports it though.